src/components/utils/CheckIn.tsx

```diff
@@ -0,0 +1,339 @@
+import { WhereToVote } from "@mui/icons-material";
+import {
+    Alert,
+    AlertTitle,
+    Collapse,
+    Fab,
+    FormControl,
+    FormControlLabel,
+    FormHelperText,
+    InputLabel,
+    MenuItem,
+    Modal,
+    Select,
+    Stack,
+    Switch,
+    Tooltip,
+} from "@mui/material";
+import Button from "@mui/material/Button";
+import React, { useCallback, useEffect, useMemo, useRef, useState } from "react";
+
+import ModalWrapper from "@/components/modals/ModalWrapper";
+import { post } from "@/lib/helpers/requests";
+import { getStoredCheckInConfiguration, storeCheckInConfiguration } from "@/lib/helpers/storage";
+import { useChainUser } from "@/lib/hooks/useChainUser";
+import { useUser } from "@/lib/hooks/useUser";
+import { CheckInTerminal, RezervoChain } from "@/types/chain";
+
+export interface CheckInLocation {
+    id: string;
+    name: string;
+    terminals: CheckInTerminal[];
+}
+
+enum ValidationError {
+    LOCATION_MISSING = "LOCATION_MISSING",
+    TERMINAL_MISSING = "TERMINAL_MISSING",
+}
+enum CheckInResult {
+    SUCCESS = "SUCCESS",
+    FAILURE = "FAILURE",
+}
+
+const checkInMessages = [
+    "Vi håper du får en god økt! 🤸‍♀️",
+    "Gi alt på trening i dag! 💪",
+    "Ha det gøy på trening! 💃",
+    "Kos deg med dagens trening! 🤗",
+];
+
+function getRandomCheckInMessage() {
+    const randomIndex = Math.floor(Math.random() * checkInMessages.length);
+    return checkInMessages[randomIndex] ?? "";
+}
+
+function filterAvailableCheckInLocations(chain: RezervoChain, selectedLocationIds: string[]) {
+    return chain.branches.flatMap((branch) =>
+        branch.locations
+            .filter(
+                (location) => selectedLocationIds.includes(location.identifier) && location.checkInTerminals.length > 0,
+            )
+            .map(
+                (location): CheckInLocation => ({
+                    id: location.identifier,
+                    name: `${chain.profile.name} ${location.name}`,
+                    terminals: location.checkInTerminals,
+                }),
+            ),
+    );
+}
+
+// TODO after merged: Check whether you can actually check in to a class *without* printing a ticket, aka. that you are registered as attending even without printing a ticket
```

Comment:
Do we need this in the code?

Reply:
I guess not, was just a note for me 🤠 Removed it